### PR TITLE
[OTA-2641] Keep some info about deleted device in a separate table

### PR DESCRIPTION
**Pull request by nkly**
Fixes [OTA-2641](https://saeljira.it.here.com/browse/OTA-2641)

### CODE REVIEW

**Comment by doanac** [comment]
Nice idea. I've recently had to add a hack into our deployment to deal with this:

 https://github.com/foundriesio/ota-api/commit/6a494d6d981c986c2e76938c360cd6981218cddb

One thing I see missing here is code to prevent a deleted device from being re-created. Is that something that could be added to this PR?

**Comment by simao** [comment]
@doanac It seems like recreating a deleted device could be a useful use case... Why do you think it should not be allowed ?

**Comment by doanac** [comment]
@simao - i apologize because I no longer have the details on me, but we hit an issue where someone added their deleted device back by accident and things "didn't work". I don't remember exactly what didn't work, but we just decided to block it from happening. Maybe the real "fix" is to understand what broke when it was re-added though.

File: src/main/resources/db/migration/V30__add_deleted_devices.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE DeletedDevice (
+    namespace CHAR(200) NOT NULL,
+    device_uuid char(36) NOT NULL,
```
Comment (houcros) [inline src/main/resources/db/migration/V30__add_deleted_devices.sql]: My OCD is crying on seeing this lower-case `char` mingled with all the upper-cases 🙈

File: src/main/resources/db/migration/V30__add_deleted_devices.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    device_id varchar(200) NOT NULL,
```
Comment (houcros) [inline src/main/resources/db/migration/V30__add_deleted_devices.sql]: I think we could benefit from a `created_at` column here, to know when the device was deleted.

File: src/main/resources/db/migration/V30__add_deleted_devices.sql (hunk 1 above)
Comment (nkly) [reply src/main/resources/db/migration/V30__add_deleted_devices.sql]: Apparently that's the convention used in this service :neutral_face:

**Pull request merged**